### ISSUE TITLE
hipGetLastError(); -Wunused-result

### ISSUE THREAD

**Issue by andrewcorrigan** [ROCmSoftwarePlatform/rocThrust#163]
rocThrust is triggering a warning in a call to [`hipGetLastError`](https://github.com/ROCmSoftwarePlatform/rocThrust/blob/develop/thrust/system/hip/detail/malloc_and_free.h#L71)
```
/rocthrust/thrust/system/hip/detail/malloc_and_free.h:71:9: warning: ignoring return value of function declared with 'nodiscard' attribute [-Wunused-result]
        hipGetLastError(); // Clear global hip error state.
        ^~~~~~~~~~~~~~~
```

This is with:
```
% clang --version
clang version 12.0.0 (/llvm/llvm-project/clang 009181e63cd6a46681319335b18180d91d05f241)
```

**Comment by andrewcorrigan** [comment]
Same with calls at [thrust/system/hip/detail/util.h:189](https://github.com/ROCmSoftwarePlatform/rocThrust/blob/develop/thrust/system/hip/detail/util.h#L189) and [thrust/system/hip/detail/util.h:221](https://github.com/ROCmSoftwarePlatform/rocThrust/blob/develop/thrust/system/hip/detail/util.h#L221)
